lib/launcher_download.js

```diff
@@ -44,11 +43,12 @@ class LauncherDownload {
   getVersionInfos (version) {
     if (this.versionsInfos[version]) { return Promise.resolve(this.versionsInfos[version]) }
     return this.getVersionsList()
-      .then(versionsList => {
+      .then(versionsList => new Promise((resolve, reject) => {
         const versionInfos = versionsList.versions.find(({ id }) => id === version)
         const versionUrl = versionInfos.url
-        return etagDownload(versionUrl, this.mcPath + '/versions/' + version + '/' + version + '.json')
-      })
+        const path = `${this.mcPath}/versions/${version}/${version}.json`
+        downloadFile(versionUrl, path, null, null).then(() => resolve(path)).catch(err => reject(err))
```

Comment:
That's not right, you should just return
Why do you need to ignore the error ?

Reply:
Ah no i see you're not ignoring it
Then why do you need to do new Promise ?

Reply:
We need to reaolve the path for the next .then, so I cant just return the download file.

Reply:
So you can do something like return download(...).then (() => path)

Reply:
I understand now, done in the latest commit